### PR TITLE
Update SDK to align with Android SDK

### PR DESCRIPTION
**Pull request by kiootic**
@roxk Please help to check if there are any missing APIs that Android SDK implemented.

### CODE REVIEW

File: Sources/Auth/Authgear.swift [hunk 1]
```diff
@@ -53,13 +53,20 @@ public struct AuthorizeResponse {
...
+}
+
 public enum AuthgearPage: String {
     case settings = "/settings"
     case identity = "/settings/identities"
 }
 
 public protocol AuthgearDelegate: AnyObject {
     func onRefreshTokenExpired()
+    func onSessionStateChanged()
```
Comment (roxk) [inline Sources/Auth/Authgear.swift:69]: 1. Seems passing in session state in the delegate would be better? Or is it common in ios to not pass in the data?
2. What is the best practice for ios's delegate in multiple threading scenarios...? The expired/logged in might be called in background thread. Android needs a bunch of handler for each delegate/methods
3. Is `ChangeReason` going to be in another PR?

**Review by roxk — CHANGES_REQUESTED** [review]
@kiootic I think only https://github.com/authgear/authgear-sdk-android/issues/11 is left, which is to add `isAnonymous` to user info.

IIRC louis says there is another issue that says should add `fetchUserInfo` as public API as well.

File: Sources/Auth/Authgear.swift (hunk 1 above)
Comment (kiootic) [reply Sources/Auth/Authgear.swift:69]: 1. I'm not familiar with idiomatic iOS API design too, but I do agree that passing in a snapshot of the state is better. Let me change that.
2. Usually delegate functions are called in main thread. It seems the synchronization is not yet sufficient regarding to internal states, let me fix that as well.
3. Yes.

Comment (louischan-oursky) [reply Sources/Auth/Authgear.swift]: The common practice is passing self as the first argument.

```swift
func onSessionStateChange(container: Authgear)
// Assume container.sessionState is public
```

For example https://developer.apple.com/documentation/uikit/uiscrollviewdelegate

Comment (kiootic) [reply Sources/Auth/Authgear.swift]: BTW, it's common to see names like `didChange`/`willChange` in iOS API, should we change to use these terms in delegate function names as well?

Comment (louischan-oursky) [reply Sources/Auth/Authgear.swift]: If we follow the convention strictly, it should be 

```swift
func authgearDidChangeSessionState(container: Authgear)
```

Comment (kiootic) [reply Sources/Auth/Authgear.swift]: I think `authgearSessionStateDidChange` sounds better, since we usually see `nounDidVerb` pattern (e.g. `viewDidAppear`, `textDidChange`)

**Review by louischan-oursky — APPROVED** [review]
